File: backend/src/models/SupportTicket.js
import mongoose from 'mongoose';

const supportTicketSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  customerName: { type: String, required: true },
  phone: { type: String, required: true },
  subject: { type: String, required: true },
  description: { type: String, required: true },
  status: { type: String, enum: ['Open', 'In Progress', 'Resolved', 'Closed'], default: 'Open', index: true },
  adminNotes: { type: String, default: '' }
}, {
  timestamps: true
});

export const SupportTicket = mongoose.model('SupportTicket', supportTicketSchema);

File: app/src/main/java/com/housieshopping/app/domain/model/SupportTicket.kt
package com.housieshopping.app.domain.model

enum class TicketStatus {
    OPEN,
    IN_PROGRESS,
    RESOLVED,
    CLOSED
}

data class SupportTicket(
    val id: String,
    val ticketNumber: String,
    val subject: String,
    val category: String,
    val description: String,
    val status: TicketStatus,
    val createdAt: String,
    val updatedAt: String
)

data class FAQItem(
    val id: String,
    val question: String,
    val answer: String,
    val category: String
)
